packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1328,13 +1331,17 @@ export class ContainerRuntime
 				compressionOptions,
 				maxBatchSizeInBytes: runtimeOptions.maxBatchSizeInBytes,
 				disablePartialFlush: disablePartialFlush === true,
+				disableBatchRebasing: disableBatchRebasing === true,
 			},
 			logger: this.mc.logger,
 			groupingManager: opGroupingManager,
 			getCurrentSequenceNumbers: () => ({
 				referenceSequenceNumber: this.deltaManager.lastSequenceNumber,
 				clientSequenceNumber: this._processedClientSequenceNumber,
 			}),
+			reSubmit: this.reSubmit.bind(this),
+			opReentrancy: () => this.ensureNoDataModelChangesCalls > 0,
```

Comment:
This is probably fine as a starting point (given that we have already some structure here).
But I'd think that if I were building it from scratch, I'd do it differently, i.e.
- Make decisions at the end of processing batch of ops
- If we are offline, noop, as we already will go through rebase
- if we are online, but there are no pending ops - noop
- otherwise, run ops through rebase.

This assumes that batches are closed on clear stack, which is not the case of immediate mode. I'd defer solving it for such mode (i.e. let's prove first that it works for explicit batched mode first).